### PR TITLE
Automate weekly pulumi/pulumi upgrade

### LINKED ISSUE #1264: Automate keeping pulumi/pkg reference updated on a weekly cadence
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Avoid P1s like https://github.com/pulumi/pulumi-awsx/issues/1262 

Keep the latest fixes flowing

### PR DESCRIPTION
**Pull request by flostadler**
This sets up a cron job to bump pulumi/pulumi versions every week. The cron job updates the dependencies, rebuilds the provider and then opens a PR. The PR is set to auto merge.

Closes #1264

### CODE REVIEW

File: .github/workflows/weekly-pulumi-update.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+name: weekly-pulumi-update
+on:
+  schedule:
+  - cron: 0 11 * * 1
```
Comment (flostadler) [inline .github/workflows/weekly-pulumi-update.yml:4]: Every monday at 11AM UTC

Comment (iwahbe) [reply .github/workflows/weekly-pulumi-update.yml:4]: If you need to explain it, why not put it in the code.
```suggestion
  # Every Monday at 11AM UTC
  - cron: 0 11 * * 1
```

File: scripts/bump-pulumi-deps.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
```
Comment (iwahbe) [inline scripts/bump-pulumi-deps.sh:1]: I'm having trouble understanding why this script isn't just
```sh
go get -u github.com/pulumi/pulumi/pkg/v3
go get -u github.com/pulumi/pulumi/sdk/v3
go mod tidy
```

**Review by iwahbe — COMMENTED** [review]
This looks like a good idea.

**Review by t0yv0 — APPROVED** [review]
Err on the side of merging, can fixup or edit as needed I think. Thank you! One day perhaps a common way to do this in ci-mgmt but it doesn't have to be now.

**Comment by iwahbe** [comment]
> Err on the side of merging, can fixup or edit as needed I think. Thank you! One day perhaps a common way to do this in ci-mgmt but it doesn't have to be now.

CC @danielrbradley who is working on unifying theses repos in ci-mgmt.

**Pull request merged**